go-controller/pkg/ovn/ovn.go

```diff
@@ -122,11 +122,20 @@ func networkStatusAnnotationsChanged(oldPod, newPod *kapi.Pod) bool {
 	return oldPod.Annotations[nettypes.NetworkStatusAnnot] != newPod.Annotations[nettypes.NetworkStatusAnnot]
 }
 
+func (oc *DefaultNetworkController) isPodScheduledinLocalZone(pod *kapi.Pod) bool {
+	if !util.PodScheduled(pod) {
+		return false
+	}
+
+	_, isLocalZoneNode := oc.localZoneNodes.Load(pod.Spec.NodeName)
```

Comment:
node watcher starts before the pod watcher.. so its expected this cache is loaded before pods are watched right? we don't want races between addnode and addpod events. I had this problem for EIP for the egressNodes watcher stuff..this cache would fluctuate and take time to get populated and simultaneously I'd get egress watch node events.